x-pack/plugins/security/server/authentication/providers/basic.test.ts

```diff
@@ -30,33 +28,47 @@ describe('BasicAuthenticationProvider', () => {
       const credentials = { username: 'user', password: 'password' };
       const authorization = generateAuthorizationHeader(credentials.username, credentials.password);
 
-      mockScopedClusterClient(mockOptions.client, sinon.match({ headers: { authorization } }))
-        .callAsCurrentUser.withArgs('shield.authenticate')
-        .resolves(user);
+      const mockScopedClusterClient = elasticsearchServiceMock.createScopedClusterClient();
+      mockScopedClusterClient.callAsCurrentUser.mockResolvedValue(user);
+      mockOptions.client.asScoped.mockReturnValue(mockScopedClusterClient);
 
       const authenticationResult = await provider.login(
-        httpServerMock.createKibanaRequest(),
+        httpServerMock.createKibanaRequest({ headers: {} }),
```

Comment:
__note:__ I do this all over the place since default request mock includes additional test header and I didn't want to verify it in the test assertions. Aaand I verify headers in assertions cause I want to make sure that `asScoped` called with the correct headers. Sinon was nicer than Jest mocks in this area - you could assign different mock implementations based on the arguments.